scripts/upgrade_admin_server.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+#
+# This script starts the upgrade of admin server to the version of system and/or
+# Cloud product.
+# The script assumes that the repositories are already correctly changed to point
+# to the new product versions. After the package upgrade is done, script initiates
+# a reboot of the machine.
+
+
+upgrade_admin_server()
+{
+    # we will need the dump for later migrating it into postgresql
+    pushd /opt/dell/crowbar_framework
+    RAILS_ENV=production
+    bin/rake db:migrate
+    bin/rake db:dump
+    popd
+
+    # Upgrade the distribution non-interactively
+    zypper --releasever 12.2 ref -f
+    zypper --non-interactive dist-upgrade -l --recommends --replacefiles
+    ret=$?
+    if [ $ret != 0 ]; then
+      # In the failed case, crowbar should tell user to check zypper logs,
+      # fix the errors and continue admin server manually,
+      echo "$ret" > /var/lib/crowbar/install/admin-server-upgrade-failed
+      return
+    fi
```

Comment:
maybe exit $ret here instead of return. and i think it is also important that we can run this script multiple times, but i think it should be ok as it is.